src/d3a/d3a_core/redis_connections/aggregator_connection.py

```diff
@@ -19,34 +22,49 @@ def set_aggregator_device_mapping(self, aggregator_device):
         self.aggregator_device_mapping = aggregator_device
         self.device_aggregator_mapping = {
             dev: aggr
-            for aggr, devices in self.aggregator_device_mapping
+            for aggr, devices in self.aggregator_device_mapping.items()
             for dev in devices
         }
 
     def is_controlling_device(self, device_uuid):
         return device_uuid in self.device_aggregator_mapping
 
-    def add_batch_event(self, device_uuid, event):
+    def _add_batch_event(self, device_uuid, event, batch_event_dict):
         aggregator_uuid = self.device_aggregator_mapping[device_uuid]
 
-        if aggregator_uuid not in self.batch_events:
-            self.batch_events[aggregator_uuid] = []
+        if aggregator_uuid not in batch_event_dict:
+            batch_event_dict[aggregator_uuid] = []
 
-        self.batch_events[aggregator_uuid].append(event)
+        batch_event_dict[aggregator_uuid].append(event)
+
+    def add_batch_market_event(self, device_uuid, event):
+        self._add_batch_event(device_uuid, event, self.batch_market_cycle_events)
+
+    def add_batch_tick_event(self, device_uuid, event):
+        self._add_batch_event(device_uuid, event, self.batch_tick_events)
+
+    def add_batch_trade_event(self, device_uuid, event):
+        self._add_batch_event(device_uuid, event, self.batch_trade_events)
+
+    def add_batch_finished_event(self, device_uuid, event):
+        self._add_batch_event(device_uuid, event, self.batch_finished_events)
```

Comment:
Is the distinction between the event types done for better debugging or does it have also a performance or stabilisation purpose?

Reply:
Dont get me wrong, I like it :-)

Reply:
I wanted to avoid the caller code to pass dict key-value pairs actually. It was way more convenient to just call the method name rather than create a dict with e.g. "event": "market"